### DISCUSSION TITLE
TrustWallet shows error Cannot read properties of undefined (reading 'type')

### DISCUSSION THREAD

**Discussion by uncleGena** [trustwallet/wallet-core#3738] [Q&A]
**Describe the bug**
When executing `transfer` on token contract, wallet opens popup with error on bottom and with disabled 'Confirm' button.
Error:
> Cannot read properties of undefined (reading 'type')

**To Reproduce**
1. [ethers.js v6.11.1](https://docs.ethers.org/v6/)
2. [trustwallet/web3-react-trust-wallet](https://github.com/trustwallet/web3-react-trust-wallet)
3. custom chain: 
 - Network name: Patex Sepolia Testnet
 - RPC URL: https://test-rpc.patex.io/
 - Chain ID: 471100
 - Currency symbol: ETH
 - Block explorer URL: https://testnet.patexscan.io/
5. custom token: 
 - address: 0x4200000000000000000000000000000000000042
 - symbol: PATEX
 
```typescript
const data = contractERC20Token.interface.encodeFunctionData('transfer', [
  someAddressSendTokensTo,
  balanceOfCustomToken
])

const tx = await signer.sendTransaction({
  to: contractERC20TokenAddress,
  data,
})
```

**Expected behavior**
Shows popup without JS error on bottom with correct data and balance of native and transfer tokens.

**Screenshots**
<img width="385" alt="image" src="https://github.com/trustwallet/wallet-core/assets/11743243/1c894aea-0df1-455b-90fb-1c7ebc1d4d35">

**Additional context**

**Comment by satoshiotomakan** [comment]
Hi @stylianos-panagakos, could you please check this issue?

**Comment by kernel1983** [comment]
![image](https://github.com/user-attachments/assets/4a9161a4-4d2e-4734-830a-238135f0a15e)

Same issue when testing in local.
